client/components/external-link/index.jsx

```diff
@@ -36,7 +36,7 @@ export default React.createClass( {
 			'has-icon': !! this.props.icon,
 		} );
 
-		const props = assign( {}, this.props, {
+		const props = assign( {}, omit( this.props, 'icon', 'iconSize' ), {
```

Comment:
Yeah, while I'd say there should be a general preference towards explicitly passing props, in cases like this where we want to treat it like an enhanced DOM anchor tag, I think it's reasonable to blindly pass props.

More pragmatic at least than trying to whitelist both [HTMLElement](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement#Properties) and [HTMLAnchorElement](https://developer.mozilla.org/en-US/docs/Web/API/HTMLAnchorElement#Properties) properties as we decide to use them.